src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -1058,17 +1058,20 @@ def extract_assay_expression(parser_obj):
 
     # Validating input combinations
     parser_dict = vars(parser_obj)
+    input_validator = InputsValidator(parser_dict=parser_dict, schema=EXTRACT_ASSAY_EXPRESSION_INPUT_ARGS_SCHEMA, error_handler=err_exit)
+    input_validator.validate_input_combination()
 
-    input_validator = InputsValidator(parser_dict=parser_dict, schema=input_arguments_validation_schemas.EXTRACT_ASSAY_EXPRESSION_INPUT_ARGS_SCHEMA, error_handler=err_exit)
-    
-    input_validator.validate()
-
-
-
+    # Validating Assay Path
+    assay_path = parser_dict.get("path")
+    project, folder_path, entity_result = resolve_existing_path(
+                assay_path
+            )
 
-    # path_validator = PathValidator(args.path)
-    # http_request_info = path_validator.get_http_request_info()
-    # input_validator.validate_cohort_list_assay(http_request_info)
+    path_validator = PathValidator(parser_dict=parser_dict, project=project, entity_result=entity_result, error_handler=err_exit)
+    path_validator.is_object_in_current_project()
+    path_validator.is_cohort_or_dataset()
+    path_validator.assert_dataset_version(3.0)
+    path_validator.cohort_list_assays_invalid_combination()
```

Comment:
@gabriela-reznik I like the granularity! Since this is the top-level use, I would recommend wrapping this sequence of 4 methods in yet another method which would be primarily exposed: `path_validator.validate()`